api/database_engine.go

```diff
@@ -87,8 +91,22 @@ func (e *EverestServer) UpgradeDatabaseEngineOperator(ctx echo.Context, namespac
 
 	// Check that this version is available for upgrade.
 	if u := dbEngine.Status.GetPendingUpgrade(req.TargetVersion); u == nil {
+		return errDBEngineUpgradeUnavailable
+	}
+
+	// Validate preflight checks.
+	preflight, err := e.getOperatorUpgradePreflight(ctx.Request().Context(), req.TargetVersion, name, namespace)
+	if err != nil {
 		return err
 	}
+	notReady := slices.ContainsFunc(pointer.Get(preflight.Databases), func(db OperatorUpgradePreflightForDatabase) bool {
+		return pointer.Get(db.PendingTask) != Ready
+	})
```

Comment:
I think it would be great to have this as a separate function, something along the lines `canUpgrade(OperatorUpgradePreflight) bool` so this can be easily reused in other places without having to figure out what needs to be checked.

Or alternatively return a third `bool` return value from `getOperatorUpgradePreflight` which would indicate if you can upgrade.

Reply:
Thanks, updated